### DISCUSSION TITLE
Is there any API to convert ECDSA signature to ASN.1 encoding?

### DISCUSSION THREAD

**Discussion by markandpathak** [Infineon/optiga-trust-m#63] [Q&A] [answered]
I am using Optiga Trust M for IoT application and really new to this ASN.1 encoding. Is there any API which can be used to convert obtained r,s component from the chip to ASN.1 encoding? I could not find such API in TRM and Wiki.

**Comment by ayushev** [comment, marked as answer]
Hello @markandpathak 
if you are refereing to the [signature](https://github.com/Infineon/optiga-trust-m/wiki/Data-format-examples#ECDSA-Signature) you get from the chip, then it is already in the ASN.1 format, some platform though require addtional tags befoer that value:
1 byte - 0x03
2 byte - size of the signature, for the NISTP256 it will be 0x42
3 byte - 0x00

**Reply by markandpathak** [reply]
@ayushev Thank you very much for the clarification. It will be nice if this information is included in the [signature](https://github.com/Infineon/optiga-trust-m/wiki/Data-format-examples#ECDSA-Signature) description to provide more clear idea to beginners about signature received from IC.
